games/xworld/maps/XWorldNav.py

```diff
@@ -7,35 +7,40 @@ class XWorldNav(XWorldEnv):
     def __init__(self, item_path):
         super(XWorldNav, self).__init__(
             item_path=item_path,
-            max_height=7,
-            max_width=7)
+            max_height=5,
+            max_width=5)
         self.curriculum = get_flag("curriculum")
+        self.current_level = 0
 
     def _configure(self):
         self.set_goal_subtrees(["animal", "fruit", "shape"])
         self.set_entity(type="agent")
 
-        num_games = self.get_num_games()
-
         ## compute world dims
-        min_dim = 3
-        max_h, max_w = self.get_dims()
+        min_dim = 2
+        max_h, max_w = self.get_max_dims()
+        n_levels = max_h - min_dim + 1
+        max_num_goals, min_num_goals = 1, 1
+        max_num_blocks, min_num_blocks = 2, 0
 
         if self.curriculum == 0:
             progress = 1.0
-            delta_dim = max_h - min_dim
+            current_dim = max_h
+            num_goals = max_num_goals
+            num_blocks = max_num_blocks
         else:
-            progress = min(float(num_games) / self.curriculum, 1.0)
-            delta_dim = int(progress * (max_h - min_dim))
-        self.set_dims(min_dim + delta_dim, min_dim + delta_dim)
+            if self.current_usage >= self.curriculum:
+                ## move to the next stage
+                self.current_level = min(self.current_level + 1, n_levels - 1)
+            current_dim = min_dim + self.current_level
+            rate = float(self.current_level) / (n_levels - 1)
+            num_goals = int(rate * (max_num_goals - min_num_goals) + min_num_goals)
+            num_blocks = int(rate * (max_num_blocks - min_num_blocks) + min_num_blocks)
 
+        self.set_dims(current_dim, current_dim)
```

Comment:
why not use current_height and current_width? otherwise we do not need height and width but just dim

Reply:
This is just a demonstration of curriculum. Of course you can do current_height and current_width, but I was too lazy to do so.